components/org.wso2.carbon.identity.conditional.auth.functions.http/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/http/util/ClientCredentialAuthConfig.java

```diff
@@ -163,10 +175,11 @@ public HttpUriRequest applyAuth(HttpUriRequest request, AuthConfigModel authConf
             if (LoggerUtils.isDiagnosticLogsEnabled()) {
                 DiagnosticLog.DiagnosticLogBuilder diagnosticLogBuilder = new
                         DiagnosticLog.DiagnosticLogBuilder(Constants.LogConstants.ADAPTIVE_AUTH_SERVICE,
-                        Constants.LogConstants.ActionIDs.RECEIVE_TOKEN);
+                        getRequestTokenActionId(getRequest()));
```

Comment:
Within the same function you seems to be storing the http request to a instance valriable and retrieving back. I'm not quite clear on the need for that while the request is already being passed as a variable

Reply:
ok I think you have been using the request of the function which invokes the client credential grant.

If that's the case, for each request the httppost or httpget functions initiate will there be an instance of this class to initiate the token request ? If that's the case storing it as an instance variable makes sense. Else if the same object is being used there will be inconsistent states

Reply:
Resolved with https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/161/commits/3552a120da28ec992f1e3428119cf1498c8e2b33